File: SHRADHAA J/ForestDS/Forest.py
#!/usr/bin/python
import json

forest=[]
node={}
	
def createNode(cs, fn, src, dest, sd, sa, ad, pd):
	node['callSign']=cs
	node['flightNo']=fn
	node['source']=src
	node['destination']=dest
	node['sch_dep']=sd
	node['sch_arr']=sa
	node['avg_delay']=ad
	node['prop_delay']=pd
	return node;

def createTree():
	tree=[]
	forest.append[tree]
	return tree;
	
def addNode(index, node):
	tempTree = forest[index]
	tempTree.append(node)
	return;
	
def getIndexOfTree(callsign):
	for i in range(0,len(forest)):
		tempTree = forest[i]
		node = tempTree[0]
		if node['callSign'] == callsign:
			return i;
	return -1;
	
	

File: SHRADHAA J/ForestDS/Forest.java
package acms;

import java.sql.Time;
import java.util.*;
import org.json.JSONArray;
import org.json.JSONObject;
import org.json.JSONException;

public class Forest {
	
	ArrayList<JSONArray> forest;
	
	Forest()
	{
		forest = new ArrayList<JSONArray>();
	}
	
	public JSONArray createTree()
	{
		JSONArray tree=new JSONArray();
		forest.add(tree);
		return tree;
	}
	public void addNode(int index, JSONObject node)
	{
		JSONArray tempTree = forest.get(index);
		tempTree.put(node);
	}
	public int getIndexOfTree(String callsign)throws JSONException
	{
		for(int i=0;i<forest.size();i++)
		{
			JSONArray tempTree = forest.get(i);
			JSONObject node = (JSONObject)tempTree.get(0);
			if(node.getString("callSign") == callsign)
				return i;	
		}
		return -1;
	}
	public JSONObject createNode(String cs, String fn, String src, String dest, Time sd, Time sa, double ad)throws JSONException
	{
		JSONObject node = new JSONObject();
		node = new JSONObject();
		node.put("callSign",cs);
		node.put("flightNo",fn);
		node.put("source",src);
		node.put("destination",dest);
		node.put("sch_dep",sd);
		node.put("sch_arr",sa);
		node.put("avg_delay",ad);
		return node;
	}
}
